src/util/http_utils.js

```diff
@@ -444,6 +449,27 @@ function make_https_request(options, body, body_encoding) {
     });
 }
 
+// Write periodically to keep the connection alive
+// TODO: Every complete above the S3_KEEP_ALIVE_WHITESPACE_INTERVAL
+// will be responded in BODY, we cannot rollback from that for pre HEADER failures
+// instread of having a fine tuned way start of sending body to make sure that we won't respond by BODY on HEADER failures
+function set_keep_alive_whitespace_interval(res) {
+    res.keep_alive_whitespace_interval = setInterval(() => {
```

Comment:
Worth adding a log print in the interval:
```suggestion
    let count = 0;
    res.keep_alive_whitespace_interval = setInterval(() => {
        count += 1;
        dbg.log0(`keep_alive_whitespace_interval headersSent=${headersSent} count=${count} `);
        if (res.headersSent) {
```